packages/SwingSet/src/kernel/kernel.js

```diff
@@ -825,6 +826,38 @@ export default function buildKernel(
     return harden(policyInput);
   }
 
+  async function processAcceptanceMessage(message) {
+    kdebug(`processAcceptanceQ ${JSON.stringify(message)}`);
+    kdebug(legibilizeMessage(message));
+    if (processQueueRunning) {
+      console.error(`We're currently already running at`, processQueueRunning);
+      assert.fail(X`Kernel reentrancy is forbidden`);
+    }
+    kernelSlog.write({ type: 'crank-start', message });
+    /** @type { PolicyInput } */
+    const policyInput = ['none'];
+    try {
+      processQueueRunning = Error('here');
+
+      kernelKeeper.addToRunQueue(message);
+
+      kernelKeeper.processRefcounts();
```

Comment:
my feeling is that we could skip this `processRefcounts` because nothing could change them during a simple queue-to-queue transfer, but it's safer to leave it in, and in the future the transfer function might cause promises to be rejected or messages to go splat, and those certainly *would* change refcounts

Reply:
I am planning on revisiting refcounts in the next PR: currently popping a `send` from the runQueue decrements refcounts, and re-increments them when re-queueing on the promise queue. Obviously that's extra work we won't usually need when popping a `send` from the outbound / acceptance queue since it will likely move to another queue (except as you say if sending to an invalid promise).